app/src/main/java/chat/rocket/android/authentication/ui/AuthenticationActivity.kt

```diff
@@ -23,18 +24,28 @@ class AuthenticationActivity : AppCompatActivity(), HasSupportFragmentInjector {
     @Inject lateinit var fragmentDispatchingAndroidInjector: DispatchingAndroidInjector<Fragment>
     @Inject lateinit var presenter: AuthenticationPresenter
     val job = Job()
+    val PREF_FIRST_TIME_USER = "first_time_user"
 
     override fun onCreate(savedInstanceState: Bundle?) {
         AndroidInjection.inject(this)
         setContentView(R.layout.activity_authentication)
         setTheme(R.style.AuthenticationTheme)
         super.onCreate(savedInstanceState)
 
+        var isFirstTimeUser = Utils().readSharedSetting(this,PREF_FIRST_TIME_USER, "true")!!.toBoolean()
+
         launch(UI + job) {
             val newServer = intent.getBooleanExtra(INTENT_ADD_NEW_SERVER, false)
             presenter.loadCredentials(newServer) { authenticated ->
                 if (!authenticated) {
                     showServerInput(savedInstanceState)
+                    if (isFirstTimeUser)
```

Comment:
This is getting more complicated, we should move all the logic into the Presenter, and make it call the right `view`method